src/topology/algebra/uniform_ring.lean

```diff
@@ -150,6 +151,31 @@ instance : comm_ring (completion R) :=
       (assume a b, by rw [← coe_mul, ← coe_mul, mul_comm]),
  ..completion.ring }
 
+instance ring_completion.algebra : algebra R (completion R) :=
+(uniform_space.completion.coe_ring_hom : R →+* (completion R)).to_algebra
```

Comment:
This probably produces a diamond with the nat-smul action (I think `ring_hom.to_algebra` always does)

Reply:
Not precisely: `ℕ` and `ℤ` don't have the instances required of `R` here. The `algebra'` instance below does cause a diamond, though.

Reply:
I think that was what I meant to comment on!